### PR TITLE
fix(js): Adds Select_mfa_type to MFA challenges

### PR DESCRIPTION
**Pull request by helgabalashova**
#### Description of changes:
Adds `SELECT_MFA_TYPE` for `Auth.signIn` challenges. Replace QRCode with QRCodeCanvas in code snipped for setting up TOTP since QRCode component is deprecated in the most recent version of qrcode.react library

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
      _ref: MDX: `[link](https://link.com)`
            HTML: `<a href="https://link.com">link</a>`_
            
### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by abdallahshaban557 — DISMISSED** [review]
LGTM

File: src/fragments/lib/xr/getting-started.mdx [hunk 1]
```diff
@@ -1,3 +1,10 @@
+<Callout warning>
```
Comment (jimblanc) [inline src/fragments/lib/xr/getting-started.mdx:1]: Why are there XR changes in this PR?

Comment (helgabalashova) [reply src/fragments/lib/xr/getting-started.mdx]: For some reason, it brought my old commit here. but I removed it

**Review by jimblanc — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by timngyn — APPROVED** [review]

**Pull request merged**